### PR TITLE
Convert project to use Java 1.7

### PR DESCRIPTION
**Pull request by BeeHiveJava**

### CODE REVIEW

**Comment by BeeHiveJava** [comment]
retest this please

File: network/src/main/java/nl/jft/network/message/MessageHandlerChainSet.java [hunk 1]
```diff
@@ -27,11 +27,21 @@
         Objects.requireNonNull(handler);
 
         synchronized (chains) {
-            MessageHandlerChain<M> chain = (MessageHandlerChain<M>) chains.computeIfAbsent(handler.getType(), MessageHandlerChain::new);
+            MessageHandlerChain<M> chain = getMessageHandlerChain(handler.getType());
             chain.addMessageHandler(handler);
         }
     }
 
+    private <M extends Message> MessageHandlerChain<M> getMessageHandlerChain(Class<M> type) {
```
Comment (TehMessiah) [inline network/src/main/java/nl/jft/network/message/MessageHandlerChainSet.java]: MAJOR Remove this unused private "getMessageHandlerChain" method. [rule](http://sonarqube.tehmessiah.nl/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

File: logic/src/main/java/nl/jft/logic/LogicConstants.java [hunk 2]
```diff
@@ -9,7 +9,7 @@
 
     public static final int INTERNAL_ID = -1;
 
-    public static final LocalDateTime INTERNAL_DATETIME = LocalDateTime.MIN;
+    public static final Date INTERNAL_DATETIME = new Date(Long.MIN_VALUE);
```
Comment (TehMessiah) [inline logic/src/main/java/nl/jft/logic/LogicConstants.java]: MINOR Make this member "protected". [rule](http://sonarqube.tehmessiah.nl/coding_rules#rule_key=squid%3AS2386)

**Comment by TehMessiah** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Pull request merged**